fix: refactor admin checks for orgs

sean-brydon opened the pull request:
---

devin-ai-integration[bot] committed:
fix: replace generic org admin checks with organization-specific validation

- Add MembershipRepository.isUserOrganizationAdmin helper for organization-specific validation
- Fix security vulnerability in teams/get.handler.ts where any org admin could bypass user filtering
- Fix teams/listMembers.handler.ts to validate admin status for target organization
- Fix eventTypes/create.handler.ts to check admin status for specific team's organization
- Fix teams/removeMember.handler.ts to validate admin status for target organizations
- Fix teams/inviteMember/inviteMember.handler.ts organization validation
- Fix insights/trpc-router.ts to validate admin status for user's own organization

This ensures users can only perform admin actions on organizations they actually admin,
preventing privilege escalation across organizations.

Co-Authored-By: sean@cal.com <Sean@brydon.io>

devin-ai-integration[bot] committed:
fix: address Alex's feedback - use existing UserRepository patterns and fix constructor

- Fix UserRepository constructor to default to global prisma instance
- Remove custom MembershipRepository.isUserOrganizationAdmin method
- Replace with existing UserRepository.isAdminOfTeamOrParentOrg and isAdminOrOwnerOfTeam methods
- Fix broken filtering logic in teams/get.handler.ts
- Update all affected files to use proper existing patterns

Co-Authored-By: sean@cal.com <Sean@brydon.io>

devin-ai-integration[bot] committed:
fix: remove duplicate isUserOrganizationAdmin function and restore MembershipRepository usage

- Remove isUserOrganizationAdmin from main membership repository
- Restore MembershipRepository.findUniqueByUserIdAndTeamId usage in get.handler.ts
- Update all references to use existing UserRepository patterns
- Fix ESLint errors for unused variables with disable comments
- Address Alex's feedback on PR #23006

Co-Authored-By: sean@cal.com <Sean@brydon.io>

devin-ai-integration[bot] committed:
fix: use dependency injection for UserRepository in API v2 services

- Replace direct UserRepository instantiation with injected PrismaUserRepository
- Update constructors in event-types services and atoms service
- Add PrismaUserRepository to module providers
- Addresses zomars' feedback on proper dependency injection patterns

Co-Authored-By: sean@cal.com <Sean@brydon.io>

zomars commented on apps/api/v2/src/modules/organizations/event-types/services/organizations-event-types.service.ts:
```diff
@@ -52,7 +53,11 @@ export class OrganizationsEventTypesService {
   async getUserToCreateTeamEvent(user: UserWithProfile, organizationId: number) {
-    const isOrgAdmin = await this.membershipsRepository.isUserOrganizationAdmin(user.id, organizationId);
+    const userRepository = new UserRepository(this.dbWrite.prisma);
+    const isOrgAdmin = await userRepository.isAdminOrOwnerOfTeam({
+      userId: user.id,
+      teamId: organizationId,
+    });
```
DevinAI, this is also skipping the propper DI pattern

zomars commented:
<!-- linear:replyTo:3176113691 -->

But how come it was working before? Aren't you confusing with another "credentials" variable in upper scope?

devin-ai-integration[bot] committed:
fix: complete dependency injection fixes for UserRepository in API v2

- Fix unintentional credentials variable change in event-types-atom service
- Replace direct UserRepository instantiation with injected PrismaUserRepository in organizations service
- Add PrismaUserRepository to organizations module providers
- Fix ESLint errors for unused variables and imports
- Addresses zomars' feedback on proper NestJS dependency injection patterns

Co-Authored-By: sean@cal.com <Sean@brydon.io>

zomars commented on packages/trpc/server/routers/viewer/teams/removeMember.handler.ts:
```diff
@@ -26,9 +27,15 @@ export const removeMemberHandler = async ({ ctx, input }: RemoveMemberOptions) =
-  const isOrgAdmin = ctx.user.profile?.organizationId
-    ? await isTeamAdmin(ctx.user.id, ctx.user.profile?.organizationId)
-    : false;
+  const isOrgAdmin = await Promise.all(
+    teamIds.map(async (teamId) => {
+      const userRepository = new UserRepository();
+      return await userRepository.isAdminOfTeamOrParentOrg({
+        userId: ctx.user.id,
+        teamId: teamId,
+      });
+    })
+  ).then((results) => results.some((result) => result));
```
DevinAI, can we bulk check multiple teams in a single query? This feels like it can be a bottleneck.

devin-ai-integration[bot] committed:
perf: optimize team admin checking with bulk query in removeMember

- Add isAdminOfAnyTeamOrParentOrg method to UserRepository for bulk team admin checking
- Replace N individual queries with single bulk query in removeMember.handler.ts
- Addresses zomars' performance concern about database query bottleneck

Co-Authored-By: sean@cal.com <Sean@brydon.io>

volnei commented:
Hey @sean-brydon I know it's a big change but I believe that functions `isAdminOrOwnerOfTeam` and `isAdminOfAnyTeamOrParentOrg` should belong to `MemberShipRepository`. 

My reasons to believe that:
1. It's more natural way to lookup and reuse those functions 
2. Query is primarily on memberships table.
3. Role checks depends on membership not on user.

But I'll love to hear your thoughts here.
